### ISSUE TITLE
Control's size is incorrect when in a grid column with Width="auto" and MaxWidth="100"

### ISSUE THREAD

**Issue by yll690** [dotnet/wpf#9485]
### Description

Control's size is incorrect when in a grid column with Width="auto" and MaxWidth="100"

### Reproduction Steps

1. Create new project targeting net8.0-windows
2. Add the following code to `MainWindow.xaml`
```
    <Grid Width="200">
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="auto" MaxWidth="100"/>
            <ColumnDefinition/>
        </Grid.ColumnDefinitions>
        <TextBlock Text="1111111111111111111111111" TextWrapping="Wrap"/>
        <GridSplitter Width="5" Background="Orange" HorizontalAlignment="Right"/>
    </Grid>
```
3. Run the project.

### Expected behavior

The TextBlock's width should be 100 and the text should be wrapped. If I drag the GridSplitter, The width of TextBlock becomes correct.
![image](https://github.com/user-attachments/assets/f11a0b23-0e34-4f65-86eb-753b07e8cd31)

### Actual behavior

The TextBlock's width is its desired width and the text is not wrapped.
![image](https://github.com/user-attachments/assets/447a638d-d924-4c3e-997f-3c3876cccf6b)

**Comment by lindexi** [comment]
@yll690  See @walterlv 's blog:

- Chinese: https://blog.walterlv.com/post/the-bugs-of-grid.html
- English: https://blog.walterlv.com/post/the-bugs-of-grid-en.html

**Comment by hongruiyu** [comment]
To achieve your desired effect, you only need to set the width of the TextBlock.

```XAML
<TextBlock Text="1111111111111111111111111" Width="auto" MaxWidth="100" TextWrapping="Wrap"/>
```

**Comment by waliqadri101** [comment]
Definitely it's a bug.

**Comment by yll690** [comment]
@miloush No, the other [document](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.columndefinition.maxwidth?view=windowsdesktop-8.0#remarks) says 

> When you use these values together in the same code example, the [MinWidth](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.columndefinition.minwidth?view=windowsdesktop-8.0) value takes precedence over the [MaxWidth](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.columndefinition.maxwidth?view=windowsdesktop-8.0) value, which in turn takes precedence over the [Width](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.columndefinition.width?view=windowsdesktop-8.0) value.

So definitely it shouldn't exceed 100 when `MaxWidth="100"` is set.

**Comment by miloush** [comment]
Well the column doesn't exceed it right? You can't see the control beyond 100, it is measured as unbounded and then clipped, like it said.

I am not saying this is right or intuitive, rather pointing out that it is not as clear cut and applications might depend on this (documented) behavior.
